src/lxc/network.c

```diff
@@ -2792,18 +2783,11 @@ char *lxc_ifname_alnum_case_sensitive(char *template)
 			}
 		}
 
-		for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
-			if (!strcmp(ifa->ifa_name, name)) {
-				exists = true;
-				break;
-			}
-		}
-
-		if (!exists)
+		if (if_nametoindex(name) == 0) {
 			break;
+		}
```

Comment:
The only tiny thing I have is: We don't do `{}` around single-line if statements. :)